cclib/parser/gaussianparser.py

```diff
@@ -899,19 +899,35 @@ def extract(self, inputfile, line):
             if not hasattr(self, "geotargets"):
                 self.geovalues = []
                 self.geotargets = numpy.array([0.0, 0.0, 0.0, 0.0], "d")
-
+            allconverged = True
             newlist = [0]*4
             for i in range(4):
                 line = next(inputfile)
                 self.logger.debug(line)
                 parts = line.split()
+                if "NO" in parts[-1]:
+                    allconverged = False
                 try:
                     value = utils.float(parts[2])
                 except ValueError:
                     self.logger.error("Problem parsing the value for geometry optimisation: %s is not a number." % parts[2])
                 else:
                     newlist[i] = value
                 self.geotargets[i] = utils.float(parts[3])
+            # reset some parameters that are printed each iteration if the 
+            # optimization has not yet converged. For example, etenergies 
+            # (Issue #889) and similar properties are only reported for the
+            # final step of an optimization.
+            if not allconverged:
+                if hasattr(self, "etenergies"):
```

Comment:
I recognize that all of these are likely to be present if `etenergies` is set (same for `etdips`), but I think all of these names should be kept in a list and you use `setattr`, it'll be cleaner.